pkg/sampler/cache.go

```diff
@@ -162,6 +171,54 @@ func (s *sampleCache) getAllSamples() (allSampleInstants, int) {
 	return all, count
 }
 
+func ContainerKeyFromContainerInfo(info v1.ContainerInfo) ContainerKey {
+	labels := info.Spec.Labels
+	containerName := labels[cadvisor.ContainerLabelContainerName]
+	podNamespace := labels[cadvisor.ContainerLabelPodNamespace]
+	podName := labels[cadvisor.ContainerLabelPodName]
+	podUID := labels[cadvisor.ContainerLabelPodUID]
+	return ContainerKey{
+		ContainerID:   containerName,
+		NamespaceName: podNamespace,
+		PodName:       podName,
+		PodUID:        podUID,
+		ContainerName: containerName,
+	}
+}
+
+func (s *sampleCache) fetchCAdvisorSample(samples *sampleInstants) error {
+	if samples == nil {
+		return nil
+	}
+	containers, err := s.CadvisorClient.SubcontainersInfo("kubelet", &v1.ContainerInfoRequest{})
```

Comment:
Is the kubelet the highest level in the container hierarchy? Does this include system and kubepod workloads?

Reply:
That's a good question. I am not sure of the top.

Reply:
I think we just empty string for the cgroup for the top level.